data/src/main/java/com/microsoft/azure/kusto/data/auth/AccessTokenTokenProvider.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.kusto.data.auth;
+
+import com.microsoft.azure.kusto.data.exceptions.DataClientException;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URISyntaxException;
+
+public class AccessTokenTokenProvider extends TokenProviderBase {
+    private final String accessToken;
+
+    AccessTokenTokenProvider(@NotNull String accessToken, @NotNull String clusterUrl, String authorityId) throws URISyntaxException {

```

Comment:
The authId here is redundent.
You may want to be tolerant of authId being null.

The same goes for clusterURL.

Reply:
Good catch with authId

I think the clusterUrl should be required in a CSB (even though it's only somewhat enforced currently - I brought this up in another thread elsewhere, I'll look that up). Further, in this MSAL implementation, it's useful to always have it available to include for example in exception handling (even though it's not currently used in all classes' flows).

Reply:
Do you agree/comfortable with my approach, @yogilad ?

Reply:
I don't think you should request parameters you do not use in practice